emcache/cluster.py

```diff
@@ -221,6 +252,60 @@ def node(self, memcached_host_address: MemcachedHostAddress) -> Node:
 
         raise ValueError(f"{memcached_host_address} can not be found")
 
+    async def _autodiscovery_monitor(self) -> None:
+        logger.debug(f"Autodiscovery task started")
+        while True:
+            if self._autodiscovery is None:
+                break
+
+            try:
+                await asyncio.sleep(self._autodiscovery)
+                await self.autodiscover()
+            except asyncio.CancelledError:
+                # if a cancellation is received we stop monitoring for cluster changes
+                break
+
+            except Exception:
+                logger.exception(f"Autodiscover raised an exception, continuing processing ")
+
+        logger.debug("Autodiscovery task stopped")
+
+    async def _get_autodiscovered_nodes(self) -> tuple[bool, int, list[tuple[str, str, int]]]:
+        if self._closed:
+            raise RuntimeError("Emcache client is closed")
+
+        node = random.choice(self._original_nodes)
+        async with OpTimeout(self._timeout, self._loop):
+            async with node.connection() as connection:
+                return await connection.autodiscovery()
+
+    async def autodiscover(self) -> None:
+        autodiscovery, version, nodes = await self._get_autodiscovered_nodes()
+        if not autodiscovery:
+            logger.warning("autodiscovery request failed")
+            return
+
+        if version == self._autodiscover_version:
+            return
+
+        if not nodes:
+            logger.error("Node list received from autodiscovery is empty!")
+            return
+
+        new_nodes: list[Node] = []
+        for host, ip, port in nodes:
+            try:
+                new_nodes.append(self.node(MemcachedHostAddress(host, port)))
+            except ValueError:
+                new_nodes.append(Node(MemcachedHostAddress(host, port), *self._new_node_options))
+
+        self._healthy_nodes = new_nodes
```

Comment:
Not sure if we should go and add all of them as healthy nodes, for the ones that already existed in previous versions and they were marked as unhealthy IMO we should keep them as unhealthy and only having them recovered back if by them selves mark them as healthy again.

Another option, is that instead of rebuilding the whole list we just add the new ones and remove the no longer available, while we keep the ones that were already there but without adding them again. This should allow us to not have to change any healthy status